prow/tide/tide.go

```diff
@@ -312,16 +360,38 @@ func NewController(ghcSync, ghcStatus github.Client, mgr manager, cfg config.Get
 	mergeChecker := newMergeChecker(cfg, ghcSync)
 
 	ctx := context.Background()
-	sc, err := newStatusController(ctx, logger, ghcStatus, mgr, gc, cfg, opener, statusURI, mergeChecker, usesGitHubAppsAuth)
+	// Shared fields
+	dontUpdateStatus := &threadSafePRSet{}
+	statusUpdate := &statusUpdate{}
+	newPoolPending := make(chan bool)
+	sc, err := newStatusController(ctx, logger, ghcStatus, mgr, gc, cfg, opener, statusURI, mergeChecker, usesGitHubAppsAuth, dontUpdateStatus, statusUpdate, newPoolPending)
 	if err != nil {
 		return nil, err
 	}
 	go sc.run()
 
-	return newSyncController(ctx, logger, ghcSync, mgr, cfg, gc, sc, hist, mergeChecker, usesGitHubAppsAuth)
+	syncCtrl, err := newSyncController(ctx, logger, ghcSync, mgr, cfg, gc, hist, mergeChecker, usesGitHubAppsAuth, dontUpdateStatus, statusUpdate, newPoolPending)
+	if err != nil {
+		return nil, err
+	}
+	return &Controller{syncCtrl: syncCtrl, statusCtrl: sc}, nil
 }
 
-func newStatusController(ctx context.Context, logger *logrus.Entry, ghc githubClient, mgr manager, gc git.ClientFactory, cfg config.Getter, opener io.Opener, statusURI string, mergeChecker *mergeChecker, usesGitHubAppsAuth bool) (*statusController, error) {
+func newStatusController(
+	ctx context.Context,
+	logger *logrus.Entry,
+	ghc githubClient,
+	mgr manager,
+	gc git.ClientFactory,
+	cfg config.Getter,
+	opener io.Opener,
+	statusURI string,
+	mergeChecker *mergeChecker,
+	usesGitHubAppsAuth bool,
+	dontUpdateStatus *threadSafePRSet,
+	statusUpdate *statusUpdate,
+	newPoolPending chan bool,
```

Comment:
We can put the other shared fields in the statusUpdate struct.

Reply:
agreed, done